packages/connectivity/src/scp-cf/destination/destination-service-types.ts

```diff
@@ -239,13 +239,44 @@ export type DestinationRetrievalOptions = CachingOptions & {
 };
 
 /**
- * Typeguard to find if object is a Destination.
- * @param destination - Destination to be checked
- * @returns boolean
- * @internal
+ * Destination for HTTP request where the URL is mandatory.
+ */
+export type HttpDestination = Destination & { url: string };
+
+/**
+ * Assertion that the Destination is a HttpDestination.
+ * This method comes in handy when you retrieved a destination from the destination service and need to check if it is a HttpDestination.
+ * @param destination - Destination or HttpDestination.
+ */
+export function assertHttpDestination(
+  destination: Destination | HttpDestination
+): asserts destination is HttpDestination {
+  if (!isHttpDestination(destination)) {
+    if (!destination.url) {
+      throw new Error(
+        `The 'url' property is not set for destination ${destination.name} which is mandatory if you use it as an 'HTTP destination`
+      );
+    }
+    throw new Error(
+      `The 'type' property is  ${destination.type} instead of  HTTP for destination '${destination.name}' which is mandatory if you use it as an 'HTTP destination`
+    );
+  }
+}
+
+/**
+ * Type guard that the Destination is HTTP destination.
+ * This method comes in handy when you retrieved a destination from the destination service and need to check if it is a HttpDestination.
+ * @param destination - Destination or HttpDestination.
+ * @returns True if the destination is a HTTP destination.
  */
-export function isDestination(destination: any): destination is Destination {
-  return destination.url !== undefined;
+export function isHttpDestination(
+  destination: HttpDestination | Destination | null
+): destination is HttpDestination {
+  if (destination?.type && destination.type !== 'HTTP') {
+    return false;
+  }
+
+  return !!destination?.url;
```

Comment:
[pp] Shorter and better readable in my opinion:
```suggestion
return destination?.type === 'HTTP' && destination?.url;
```